### PR TITLE
Fix JIT use of BitScan* APIs

### PR DESCRIPTION
**Pull request by BruceForstall**
windows.h defines the BitScan* APIs with a leading underscore and includes `#define` definitions of non-underscore versions to the underscore versions.

This is annoying in the JIT, where we have a BitOperations class that uses these same names; the class members end up with leading underscores and it confuses Visual Studio source browsing.

In the JIT, `#undef` the windows.h underscore defines. Define pass-through non-underscore functions to call the actual functions. (We need to always call the non-underscore versions because that is what is defined in the PAL.) Replace usage of bitposition.h in the JIT and remove it from utilcode.h (only one other place in the CLR uses it and they already include it.)

### CODE REVIEW

File: src/coreclr/jit/compiler.cpp [hunk 1]
```diff
@@ -26,14 +26,6 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
...
 extern ICorJitHost* g_jitHost;
 
-#if defined(DEBUG)
-// Column settings for DOTNET_JitDumpIR.  We could(should) make these programmable.
-#define COLUMN_OPCODE 30
-#define COLUMN_OPERANDS (COLUMN_OPCODE + 25)
-#define COLUMN_KINDS 110
-#define COLUMN_FLAGS (COLUMN_KINDS + 32)
-#endif
-
```
Comment (BruceForstall) [inline src/coreclr/jit/compiler.cpp:36]: This is unrelated dead code removal

**Review by tannergooding — APPROVED** [review]

**Comment by BruceForstall** [comment]
Failure is a known issue.

**Comment by BruceForstall** [comment]
cc @dotnet/jit-contrib

**Comment by BruceForstall** [comment]
Significant TP improvement on Linux arm, presumably because the genLog2 implementation is better.

```
Overall (-0.14% to -0.08%)
MinOpts (-0.22% to -0.16%)
FullOpts (-0.10% to -0.07%)
```
